app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1856,13 +1862,25 @@ private void removeMarker(Place place){
 
     @Override
     public void recenterMap(fr.free.nrw.commons.location.LatLng currentLatLng) {
-        if (isPermissionDenied || currentLatLng == null) {
-            recenterToUserLocation = true;
-            checkPermissionsAndPerformAction();
-            if (!isPermissionDenied && !(locationManager.isNetworkProviderEnabled()
-                || locationManager.isGPSProviderEnabled())) {
-                showLocationOffDialog();
+        // if user has denied permission twice, then show dialog
+        if (isPermissionDenied) {
+            if (locationPermissionsHelper.checkLocationPermission(getActivity())) {
+                // this will run when user has given permission by opening app's settings
+                isPermissionDenied = false;
+                locationPermissionGranted();
+                return;
```

Comment:
Hey, @sivaraam, I agree that there was no `return;` statement in the proposed change, but while testing, I found that even though after the `locationPermissionGranted()` is invocated and the user's current location is found, the `currentLatLng` in the `recenterMap()` has the default value only, so the map moves back to the default location instead of using the current location of the user. Also, we call `recenterMarkerToPosition()` in `locationPermissionGranted()` which does the same job as `addCurrentLocationMarker()` but uses the updated location which solves the problem of `recenterMap()` using the old/default location.

Reply:
Ok. Thanks for the clarification. But that makes me wonder the actual purpose of the rest of the code in the `recenterMap` method. If `locationPermissionGranted` itself is already handling all necessary code flows, do you think the rest of the code in the `recentMap` that appears after the if conditions are not really necessary? In the sense that we could refactor the method to completely remove them? 🤔 

Or are there specific cases where you think it would be helpful?

Reply:
I think there are some portions of recenterMap, which may be required, such as those setting and resetting `setSearchThisAreaButtonVisibility()` to true or false. I'll check and test this once and confirm.

Reply:
Hey @sivaraam, I removed the redundant code lines:
`addCurrentLocationMarker(currentLatLng);
binding.map.getController()
            .animateTo(new GeoPoint(currentLatLng.getLatitude(), currentLatLng.getLongitude()));` 
            
Honestly, I couldn't fully understand what the lines after these ( i.e. those setting `setSearchThisAreaButtonVisibility()` to true/false ), do and I tested Nearby without them and it seemed to work fine. But I'm not sure if we should remove them.

Reply:
Thank you @ShashwatKedia. I just pushed a comment mentioning the potential redundant code in [this commit](https://github.com/ShashwatKedia/apps-android-commons/commit/70bd771370ab05330196ad19966c627abe8c518c). We can explore the same and remove it when feasible.